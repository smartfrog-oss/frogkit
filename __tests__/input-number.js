import { mount } from 'vue-test-utils'
import InputNumber from '@/components/input-number'

describe('Input component', () => {

  it('should render component and match snapshot', () => {
    const wrapper = mount(InputNumber)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render small input', () => {
    const wrapper = mount(InputNumber, { propsData: { size : 'small' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render big input', () => {
    const wrapper = mount(InputNumber, { propsData: { size : 'big' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should disable decrease button', () => {
    const wrapper = mount(InputNumber, { propsData: { min : 1, value: 1, max: 5 } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should disable increase button', () => {
    const wrapper = mount(InputNumber, { propsData: { min : 1, value: 5, max: 5 } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should decrease input', () => {
    const wrapper = mount(InputNumber, { propsData: { min : 1, value: 3, max: 5 } })
    wrapper.findAll('button').at(0).trigger('click')
    expect(wrapper.emitted().input[0]).toEqual([2])
  })

  it('should increase input', () => {
    const wrapper = mount(InputNumber, { propsData: { min : 1, value: 3, max: 5 } })
    wrapper.findAll('button').at(1).trigger('click')
    expect(wrapper.emitted().input[0]).toEqual([4])
  })

})
