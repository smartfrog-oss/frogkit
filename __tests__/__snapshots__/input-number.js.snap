// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should disable decrease button 1`] = `
<flex class="fk-number-input"><button color="secondary" disabled="disabled" class="fk-number-input__button--small">
    <icon icon="minus" size="small"></icon>
  </button> <input type="number" value="1" aria-label="quantity" class="fk-number-input__input--small"> <button color="secondary" class="fk-number-input__button--small">
    <icon icon="plus" size="small"></icon>
  </button></flex>
`;

exports[`Input component should disable increase button 1`] = `
<flex class="fk-number-input"><button color="secondary" class="fk-number-input__button--small">
    <icon icon="minus" size="small"></icon>
  </button> <input type="number" value="5" aria-label="quantity" class="fk-number-input__input--small"> <button color="secondary" disabled="disabled" class="fk-number-input__button--small">
    <icon icon="plus" size="small"></icon>
  </button></flex>
`;

exports[`Input component should render big input 1`] = `
<flex class="fk-number-input"><button color="secondary" disabled="disabled" class="fk-number-input__button--big">
    <icon icon="minus" size="big"></icon>
  </button> <input type="number" value="1" aria-label="quantity" class="fk-number-input__input--big"> <button color="secondary" class="fk-number-input__button--big">
    <icon icon="plus" size="big"></icon>
  </button></flex>
`;

exports[`Input component should render component and match snapshot 1`] = `
<flex class="fk-number-input"><button color="secondary" disabled="disabled" class="fk-number-input__button--small">
    <icon icon="minus" size="small"></icon>
  </button> <input type="number" value="1" aria-label="" class="fk-number-input__input--small"> <button color="secondary" class="fk-number-input__button--small">
    <icon icon="plus" size="small"></icon>
  </button></flex>
`;

exports[`Input component should render small input 1`] = `
<flex class="fk-number-input"><button color="secondary" disabled="disabled" class="fk-number-input__button--small">
    <icon icon="minus" size="small"></icon>
  </button> <input type="number" value="1" aria-label="quantity" class="fk-number-input__input--small"> <button color="secondary" class="fk-number-input__button--small">
    <icon icon="plus" size="small"></icon>
  </button></flex>
`;
